File: android-app/app/src/main/java/com/example/fingerprintdemo/rest/Purchase.java
package com.example.fingerprintdemo.rest;

public class Purchase implements SignableRequestPayload {

    private final String itemName;
    private final String deliveryAddress;
    private final String token;
    private final long timestamp;

    public Purchase(String itemName, String deliveryAddress, String token) {
        this.itemName = itemName;
        this.deliveryAddress = deliveryAddress;
        this.token = token;
        this.timestamp = System.currentTimeMillis();
    }

    @Override
    public String concatenatedValuesForSigning() {
        return itemName
                + "|"
                + deliveryAddress
                + "|"
                + token
                + "|"
                + timestamp;
    }
}

File: demo-backend/src/main/kotlin/com/example/demobackend/Purchase.kt
package com.example.demobackend

class Purchase : SignableRequestPayload {
    lateinit var itemName: String
    lateinit var deliveryAddress: String
    lateinit var token: String

    override var timestamp: Long = 0

    override fun messageForVerification(): String {
        return arrayOf(itemName,
                deliveryAddress,
                token,
                timestamp).joinToString("|")
    }

}
